arch/arm/arm64/head.S

```diff
@@ -85,6 +104,170 @@ ENTRY(_start)
     load_paddr  x0, __start_bss
     bl      flush_dcache_range
 
-    /* Start an infinite loop */
-    PRINT("- Infinite loop -\n")
-2:  b   2b
+    PRINT("- Setup stack -\n")
+    ldr     x1, =stack_end
+    mov     sp, x1
+
+    /* Save boot arguments */
+    ldr     x0, =boot_data
+    stp     x21, x20, [x0]
+
+    PRINT("- Jump to C world-\n")
+    b       xtf_main
+ENDFUNC(_start)
+
+ENTRY(_mmu_disable)
```

Comment:
If I am not mistaken, `ENTRY` will export the symbol. However, this function an most of the other below are not meant to be used outside of the file. So I think you want to drop `ENTRY`. 

Also, can you please document what each function does and more importantly mention whether they clobber any registers.

Reply:
I will drop ENTRY for local functions and I will add comments about clobbering